### ISSUE TITLE
Installing dependencies on CentOS 8 fails due to libyaml-devel

### ISSUE THREAD

**Issue by strzibny** [postmodern/ruby-install#383]
`libyaml-devel` is no longer in the enabled repositories.

We need to run `dnf` with `--enablerepo=PowerTools`.

**Comment by postmodern** [comment]
I'm curious how RedHat's shift to CentOS Stream might change this?

**Comment by ilikepi** [comment]
Yeah, unfortunately this _is_ the result of RedHat setting fire to everything with RHEL and CentOS.  They pushed a bunch of packages that are primarily for developers (e.g. lots of `-devel` packages) into a separate repo so they could call the whole set of them "unsupported".  In distros derived from RHEL 8 this repo is called "powertools", and in RHEL 9 it's "crb" (short for "CodeReady Builder" apparently 🤷‍♂️).  In both cases these repos are disabled by default.  I ran into this tonight as well, on AlmaLinux 9...I had to enable the "crb" repo before I could install`libyaml-devel` and `gdbm-devel`.

I tried to find related documentation specific to CentOS Stream, but lots of what's on centos.org is broken or outdated 🙄 ...they helpfully point out that CentOS 8 Stream is EOL in about 6 weeks, even though RHEL 8 will get maintenance for another 5 years.

**Comment by postmodern** [comment]
Grumble. `*shakes fist at RedHat*`. I guess we need to add some logic to `install_packages` to check if we're running on an RHEL derived distro (how many are there besides Alma Linux?) and add `--enablerepo=PowerTools` to the `dnf install` command.

**Comment by ilikepi** [comment]
> (how many are there besides Alma Linux?)

At the very least, I think this issue would apply to:

* RHEL 8/9
* CentOS 8/9 Stream
* AlmaLinux 8/9
* Oracle Linux 8/9
* Rocky Linux 8/9

I think Amazon Linux or whatever it's called is derived from RHEL as well, but I'm not sure what version it's based on or to what degree they've modified the organization of their package repositories...

**Comment by postmodern** [comment]
Closing this since CentOS has been EoLed. I do however have plans to add Alma Linux and Rocky Linux, which are popular successors to CentOS, to the CI.
